lib/Finance/Calendar.pm

```diff
@@ -102,6 +102,8 @@ sub trades_on {
     return $result;
 }
 
+Memoize::memoize('trades_on', NORMALIZER => '_normalize_on_dates');
```

Comment:
I think in `_normalize_on_dates` we also need to use `recorded_dates` or `$self`'s address, if we continue to cache instance

Reply:
Noted.